pkg/ccl/backupccl/backupinfo/manifest_handling.go

```diff
@@ -529,6 +554,50 @@ func WriteBackupLock(
 	return cloud.WriteFile(ctx, defaultStore, lockFileName, bytes.NewReader([]byte("lock")))
 }
 
+// WriteSlimBackupManifestWithMetadataSSTs writes a "slim" version of manifest
+// to `exportStore`. This version currently has the alloc heavy `Files` repeated
+// field nil'ed out, and written to an accompanying SST instead.
+func WriteSlimBackupManifestWithMetadataSSTs(
+	ctx context.Context,
+	exportStore cloud.ExternalStorage,
+	encryption *jobspb.BackupEncryptionOptions,
+	kmsEnv cloud.KMSEnv,
+	manifest *backuppb.BackupManifest,
+) error {
+	if err := writeSlimBackupManifest(ctx, exportStore, backupbase.SlimBackupManifestName, encryption,
+		kmsEnv, manifest); err != nil {
+		return errors.Wrap(err, "failed to write the slim backup manifest")
+	}
+	return errors.Wrap(WriteBackupManifestFilesSST(ctx, exportStore, encryption, kmsEnv, manifest,
+		SlimManifestFileInfoPath), "failed to write backup manfiest files SST")
+}
+
+// WriteSlimBackupManifest compresses and writes a slim version of the
+// BackupManifest `desc` to `exportStore`.
+//
+// A slim backup manifest has the following repeated fields of the
```

Comment:
The structure of this comment seems to suggest that we might later have other fields nil'ed out in a "slim" manifest, but I think that would in fact be a new, separate, slim*er* manifest, since we'd need to keep writing these simply slim (just not files) manifests during the transition, just like we do now. I.e. I think we'll drop regular manifests and then write slim plus slimmer, then drop slim and writer slimmer and even-slimmer, etc, rather than have full vs a very slim "slim".

This makes me feel like we should maybe actually s/slim/external-filelist/ now, instead of introducing the word "slim" when it won't really be "slim" compared to what is coming later?

Reply:
good point, changed.